source/simulator/core.cc

```diff
@@ -989,6 +989,8 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::setup_dofs ()
   {
+    signals.edit_parameters(*this, parameters);
+
```

Comment:
There is a reasoning: doing it at the end of ```parse_parameters()``` means it is only called once. If I want to do some editing at other times, it needs to be in the main loop. 

I put it in  ```setup_dofs()``` because in many cases edits to the parameters only take effect when this function is called.  For instance, setting up the boundary conditions happens here, so in the use case that I've tested, this is the most appropriate place for it.

This is certainly a hack, but my understanding that this is what the signals/slots are for?